diff --git a/src/cmake/build_yaml-cpp.cmake b/src/cmake/build_yaml-cpp.cmake
index 539773726..70ea548a4 100644
--- a/src/cmake/build_yaml-cpp.cmake
+++ b/src/cmake/build_yaml-cpp.cmake
@@ -27,6 +27,7 @@ build_dependency_with_cmake(yaml-cpp
         -D YAML_CPP_BUILD_TOOLS=OFF
         -D YAML_CPP_BUILD_CONTRIB=OFF
         -D YAML_BUILD_SHARED_LIBS=${yaml-cpp_BUILD_SHARED_LIBS}
+        "-D CMAKE_CXX_FLAGS=-include cstdint"
         -D CMAKE_INSTALL_LIBDIR=lib
         -D CMAKE_POLICY_VERSION_MINIMUM=3.5
     )
diff --git a/src/cmake/build_OpenColorIO.cmake b/src/cmake/build_OpenColorIO.cmake
index 3af66f011..275a18c01 100644
--- a/src/cmake/build_OpenColorIO.cmake
+++ b/src/cmake/build_OpenColorIO.cmake
@@ -16,7 +16,7 @@ set_cache (OpenColorIO_BUILD_SHARED_LIBS  OFF
 # it all work with the static dependencies, it just makes things complicated
 # downstream.
 
-checked_find_package(pystring VERSION_MIN 1.1.3)
+checked_find_package(pystring VERSION_MIN 1.1.3 BUILD_LOCAL missing)
 checked_find_package(expat REQUIRED VERSION_MIN 2.6)
 checked_find_package(yaml-cpp REQUIRED VERSION_MIN 0.8.0)
 checked_find_package(minizip-ng REQUIRED VERSION_MIN 4.0.10)
